### PR TITLE
Smart color handling

### PR DESCRIPTION
**Pull request by doegox**
* if on Windows, no color, as usual
* if on Linux, color only on real term
  * no color when > foo
  * no color in the logfile

@iceman1001 please test it on mingw to make sure I didn't break anything.

### CODE REVIEW

File: common/commonutil.h [hunk 1]
```diff
@@ -53,4 +53,11 @@ void lsl(uint8_t *data, size_t len);
 int32_t le24toh(uint8_t data[3]);
 void htole24(uint32_t val, uint8_t data[3]);
 
+# define _BLUE_(s) "\x1b[34m" s "\x1b[0m "
+# define _RED_(s) "\x1b[31m" s "\x1b[0m "
+# define _GREEN_(s) "\x1b[32m" s "\x1b[0m "
+# define _YELLOW_(s) "\x1b[33m" s "\x1b[0m "
+# define _MAGENTA_(s) "\x1b[35m" s "\x1b[0m "
+# define _CYAN_(s) "\x1b[36m" s "\x1b[0m "
```
Comment (iceman1001) [inline common/commonutil.h]: use the guards,  it doesn't hurt.

**Comment by iceman1001** [comment]
ok,  However I think you can use guards on the defines still.
No sure if the if-case in even packaget received is a good way to place.  Rather only do that once per session, instead of every package.,

**Comment by doegox** [comment]
Defined can't be guarded for armsrc, you might compile image on windows and use it later on Linux

**Comment by doegox** [comment]
Yes test for color or not could be put somewhere in global var from proxmark.c

**Comment by iceman1001** [comment]
Looking good so far. No colors in TERM or log file.

**Comment by iceman1001** [comment]
flasher doesn't clean ansi ...

```
pm3 ~$ client/flasher.exe com9 -b bootrom/obj/bootrom.elf  armsrc/obj/fullimage.elf
←[34mLoading ELF file←[0m ←[33mbootrom/obj/bootrom.elf←[0m
Loading usable ELF segments:
0: V 0x00100000 P 0x00100000 (0x00000200->0x00000200) [R X] @0x94
1: V 0x00200000 P 0x00100200 (0x00000d90->0x00000d90) [RWX] @0x298

←[34mLoading ELF file←[0m ←[33marmsrc/obj/fullimage.elf←[0m
Loading usable ELF segments:
0: V 0x00102000 P 0x00102000 (0x0003b130->0x0003b130) [R X] @0x94
1: V 0x00200000 P 0x0013d130 (0x000013f0->0x000013f0) [RW ] @0x3b1c4
Note: Extending previous segment from 0x3b130 to 0x3c520 bytes

[+] Waiting for Proxmark3 to appear on com9
.←[32mFound←[0m
←[34mEntering bootloader...←[0m
(Press and release the button only to abort)
[+] Waiting for Proxmark3 to appear on com9
............. ←[32mFound←[0m

←[34mFlashing...←[0m
Writing segments for file: bootrom/obj/bootrom.elf
 0x00100000..0x001001ff [0x200 / 1 blocks].←[32mOK←[0m
 0x00100200..0x00100f8f [0xd90 / 7 blocks].......←[32mOK←[0m

Writing segments for file: armsrc/obj/fullimage.elf
 0x00102000..0x0013e51f [0x3c520 / 483 blocks]..............................................................................
............................................................................................................................
............................................................................................................................
.............................................................................................................←[32mOK←[0m

←[34mResetting hardware...←[0m
←[34mAll done.←[0m

Have a nice day!
```

**Comment by doegox** [comment]
fixed

**Comment by iceman1001** [comment]
I also get   STDIN un expected end  when piping
```
pm3 ~/client$ tail  proxmark3.log
[=] Communicating with PM3 over USB-CDC
pm3 --> hf 14a i
 UID : B6 50 B9 F3
ATQA : 00 04
 SAK : 08 [2]
TYPE : NXP MIFARE CLASSIC 1k | Plus 2k SL1 | 1k Ev1
[=] proprietary non iso14443-4 card found, RATS not supported
[=] Answers to magic commands: NO
[+] Prng detection: WEAK
[!!] STDIN unexpected end, exit...
pm3 ~/client$ echo 'hf 14a i' | proxmark3 com9
```

**Pull request merged**